File: java/picocontainer/src/test/org/picocontainer/defaults/DefaultComponentRegistryTestCase.java
/*****************************************************************************
 * Copyright (C) PicoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the LICENSE.txt file.                                                     *
 *                                                                           *
 * Idea by Rachel Davies, Original code by Stacy Curl                        *
 *****************************************************************************/

package org.picocontainer.defaults;

import junit.framework.TestCase;
import org.picocontainer.PicoRegistrationException;
import org.picocontainer.ComponentAdapter;
import org.picocontainer.testmodel.SimpleTouchable;
import org.picocontainer.testmodel.Touchable;

public class DefaultComponentRegistryTestCase extends TestCase {
	private DefaultPicoContainer picoContainer;

	protected void setUp() throws Exception {
		picoContainer = new DefaultPicoContainer();
	}

	public void testRegisterComponent() throws PicoRegistrationException {
		ComponentAdapter componentSpecification = createComponentAdapter();

		picoContainer.registerComponent(componentSpecification);

		assertTrue(picoContainer.getComponentAdapters().contains(componentSpecification));
	}

	public void testUnregisterComponent() throws PicoRegistrationException {
		ComponentAdapter componentSpecification = createComponentAdapter();

		picoContainer.registerComponent(componentSpecification);

		picoContainer.unregisterComponent(Touchable.class);

		assertFalse(picoContainer.getComponentAdapters().contains(componentSpecification));
	}

	private ComponentAdapter createComponentAdapter() throws AssignabilityRegistrationException, NotConcreteRegistrationException {
		return new ConstructorComponentAdapter(Touchable.class, SimpleTouchable.class);
	}
}

File: dotnet/picocontainer/src/Test/Defaults/DefaultComponentRegistryTestCase.cs
/*****************************************************************************
 * Copyright (C) PicoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the license.html file.                                                    *
 *                                                                           *
 * Idea by Rachel Davies, Original code by Aslak Hellesoy and Paul Hammant   *
 * C# port by Maarten Grootendorst                                           *
 *****************************************************************************/

using System;
using System.Diagnostics;

using csUnit;

using PicoContainer.Defaults;
using PicoContainer.Tests.Tck;
using PicoContainer.Tests.TestModel;

namespace PicoContainer.Tests.Defaults
{
  [TestFixture]
  public class DefaultComponentRegistryTestCase
  {
    private DefaultPicoContainer componentRegistry;

    [SetUp]
    public void SetUp() 
    {
      componentRegistry = new DefaultPicoContainer();
    }

    [Test]
    public void testRegisterComponent() 
    {
      ComponentAdapter componentSpecification = createComponentAdapter();

      componentRegistry.RegisterComponent(componentSpecification);

      Assert.True(componentRegistry.HasComponentAdapter(componentSpecification));
    }

    [Test]
    public void testUnRegisterComponent() 
    {
      ComponentAdapter componentSpecification = createComponentAdapter();

      componentRegistry.RegisterComponent(componentSpecification);

      componentRegistry.UnRegisterComponent(typeof(Touchable));

      Assert.False(componentRegistry.HasComponentAdapter(componentSpecification));
    }

    [Test]
    private ComponentAdapter createComponentAdapter() 
    {
      return new DefaultComponentAdapter(typeof(Touchable), typeof(SimpleTouchable));
    }
  }
}